File: domain/src/main/java/com/github/il4enkodev/househeating/domain/interactor/metering/DeleteMetering.java
package com.github.il4enkodev.househeating.domain.interactor.metering;

import com.github.il4enkodev.househeating.domain.interactor.SchedulerSwitcher;
import com.github.il4enkodev.househeating.domain.interactor.UseCaseCompletable;
import com.github.il4enkodev.househeating.domain.repository.MeteringRepository;

import javax.inject.Inject;

import io.reactivex.Completable;

public class DeleteMetering extends UseCaseCompletable<String> {

    private final MeteringRepository meteringRepository;

    @Inject
    DeleteMetering(SchedulerSwitcher<?> schedulerSwitcher, MeteringRepository meteringRepository) {
        super(schedulerSwitcher);
        this.meteringRepository = meteringRepository;
    }

    @Override
    protected Completable source(String arguments) {
        return meteringRepository.delete(arguments);
    }
}

File: domain/src/main/java/com/github/il4enkodev/househeating/domain/interactor/metering/DeleteMetering.kt
package com.github.il4enkodev.househeating.domain.interactor.metering

import com.github.il4enkodev.househeating.domain.di.IoDispatcher
import com.github.il4enkodev.househeating.domain.interactor.UseCase
import com.github.il4enkodev.househeating.domain.repository.MeteringRepository
import kotlinx.coroutines.CoroutineDispatcher
import javax.inject.Inject

class DeleteMetering @Inject internal constructor(
    @IoDispatcher private val dispatcher: CoroutineDispatcher,
    private val meteringRepository: MeteringRepository
) : UseCase<String, Unit>(dispatcher) {

    override suspend fun execute(arguments: String) {
        meteringRepository.delete(arguments)
    }
}